codecov.yaml

```diff
@@ -1,53 +1,75 @@
+#  # `Files` or Tree are a list of files that are impacted by the pull request (coverage changes, file is new or removed).
+#  # `Header` is the top of the comment showing the two commits being merged, the change in coverage, and the diff coverage.
+#  # `Footer` is the bottom of a comment that contains the legend, last updated commits, and docs link.
+#  layout: "reach,diff,flags,tree,header,footer"
+#  # The way Codecov submits comments in your pull requests.`default` means update if exists.Otherwise, post new.
+#  behavior: default
+#  # If `require_changes` set true, comments will now only post when coverage changes.
+#  # Furthermore, if a comment already exists, and a newer commit results in no coverage
+#  # change for the entire pull, the comment will be deleted.
+#  require_changes: no
+#  # Force comments to post on pull requests even if Codecov doesn't
+#  # have coverage reports for either the base or head commit.
+#  require_base: no
+#  require_head: no
+#  # Codecov does not comment until at least 5 builds have been uploaded from the CI pipeline.
+#  # By default, codecov will post and/or update the pull request comment after it processes
+#  # each report uploaded for a particular pull request commit.
+#  # As the Antrea CI process uploads many reports to codecov, this can be confusing for
+#  # team members viewing the pull request as reports are processing.
+#  after_n_builds: 5
+#  # Carryforward Flags are designed for projects that do not upload total coverage for every commit.
+#  show_carryforward_flags: false
 
 github_checks:
+  # Specify whether to use GitHub Checks annotations or normal statuses.
   annotations: true
 
 coverage:
   status:
     patch:
       default:
+        # Choose a minimum coverage ratio that the commit must meet to be considered a success.
+        # <number> specify a target of an exact coverage number.
         target: 70%
+        # Allow the coverage to drop by X%, and posting a success status.
         threshold: 5%
+        # Use the pull request base if the commit is on a pull request. If not, the parent commit will be used.
         base: auto
+        # Only post a status to pull requests.
         only_pulls: true
-        flags:
-          - unit-tests
     project:
       default:
+        # `auto` will use the coverage from the base commit (pull request base or parent commit) coverage to compare against.
         target: auto
       antrea-unit-tests:
         target: auto
+        # Flags are a list of user defined Flags, and the impact on their coverage.
         flags:
           - unit-tests
-      antrea-integration-tests:
-        target: auto
-        flags:
-          - integration-tests
-      antrea-e2e-tests:
-        target: auto
-        flags:
-          - e2e-tests
-      antrea-kind-e2e-tests:
-        target: auto
-        flags:
-          - kind-e2e-tests
```

Comment:
Why do you remove these flags? I think our CI scripts are using these flags to upload them into different categories.

Reply:
I think there is no need to put those flags in the project status check. Maybe we can not ensure that code coverage of the e2e or the integration test is increasing in each PR. e2e and integration tests are just a complement to unit-test.
For example, if there are 10k lines in a project, the UT coverage is 80%, and 20%(2k) of those lines are hard to cover by UT but we can cover them by e2e or integration test. When the project code lines increase to 20k, 2k is just 10% of them.

Reply:
The changed code blocks just determine whether report a result in the checks. Those lines in `test-vmc.sh#L525` can certainly work. It just uploads the test coverage data with the flag mark.